File: Helloworld/src/main/java/com/example/demo1/controller/RegistrationController.java
package com.example.demo1.controller;

import java.util.List;
import java.util.Optional;


import com.example.demo1.repo.CoustmerRepo;
import com.example.demo1.model.Coustmers;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.servlet.ModelAndView;

import com.example.exception.AgeException;
import com.example.exception.PasswordException;

@Controller
public class RegistrationController {
	Logger log = LoggerFactory.getLogger(RegistrationController.class);
	@Autowired
    CoustmerRepo repo;
	@RequestMapping("/")
	public String details() {
		return "registration";
	}

	//@RequestMapping("/details")
	@RequestMapping(value="/details",produces={"application/*","text/html"},consumes={"application/*"})
	@ResponseBody
	public String details(@RequestBody Coustmers coustmers ) throws PasswordException  {
		try {
			int passwordLength = coustmers.getCpassword().length();
			
			if (passwordLength < 5) {
				log.error("criteria is not matched");
				throw new PasswordException("password doesnot match criteria");

			}
			int ageLimit = coustmers.getCage();
			if (ageLimit < 18) {
				log.error("age is below the limit,"+coustmers.getCage());
				throw new AgeException("you are below tha age criteria");
			}

			repo.save(coustmers);
		}
		
//		catch (PasswordException e) {
//			System.out.println("error" + e.getMessage());
//		}
	catch (AgeException e) {
			System.out.println("error" + e.getMessage());
	}
		
		return "registration";
	}

	@RequestMapping("/getdetails")
	public String getdetails() {

		return "viewcoustmer";
	}

//	@PostMapping("/getdetails")
//	public ModelAndView getdetails(@RequestParam int cid) {
//		ModelAndView mv = new ModelAndView("retrive");
//		Coustmers coustmers = repo.findById(cid).orElse(null);
//		mv.addObject(coustmers);
//		return mv;
//	}
	@PostMapping("/getdetails")
	public ModelAndView getdetails(@RequestParam int cid) {
		Coustmers coustmers;
		ModelAndView mv = new ModelAndView("retrive");
		List<Coustmers> CoustmerList = (List<Coustmers>) repo.findAll();
		
		//mv.addObject(coustmers);
		//coustmers.forEach(details->System.out.println(details));
		
		Optional<Coustmers> matchingCoustmer=CoustmerList.stream().filter(p->p.getCid()==(cid)).findFirst();
		if(matchingCoustmer.isPresent()) {
			coustmers= matchingCoustmer.get();
			log.info("person is found"+ coustmers.getCusername());
			mv.addObject(coustmers);
		}else
		{
			coustmers=matchingCoustmer.orElse(null);
			log.info("serached uname is not found",cid);
			mv.setViewName("viewcoustmer");
		}
		return mv;
	}
}

File: Helloworld/src/main/kotlin/com/ibs/demo2/controller/RegistrationController.kt
package com.ibs.demo2.controller

import com.ibs.exceptions.AgeException
import com.ibs.exceptions.PasswordException
import com.ibs.demo2.model.Coustmers
import com.ibs.demo2.repo.CoustmerRepo
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.servlet.ModelAndView
import java.util.*
import java.util.stream.Stream


@Controller
class RegistrationController {
    var log : Logger = LoggerFactory.getLogger(RegistrationController::class.java)
    @Autowired
    lateinit var repo:CoustmerRepo
    @RequestMapping("/")
    fun details():String
    {
        return "registration"
    }
    @RequestMapping("/details")
    fun details(coustmers:Coustmers):String{
        try {
            val PasswordLength: Int = coustmers.cpassword.length
            if (PasswordLength < 5) {
                println("in sufficient length")
                log.error("in sufficient length")
                throw  PasswordException("length is less than 5")
            }
            val AgeLimit = coustmers.cage
            if (AgeLimit < 18) {
                println("age is below limit")
                log.error("age is below limit")
                throw AgeException("age is below limit")
            }


            repo.save(coustmers)
        }
       //catch(e:PasswordException){
       // println("error"+e.message)
        //}
        catch(e:AgeException){
            println("error"+e.message)
        }
        return "registration"
    }
    @RequestMapping("/getdetails")
        fun getdetails():String{
            return "viewcoustmer";
        }
       @PostMapping("/getdetails")
        fun getdetails(@RequestParam cid:Int):ModelAndView{
           val coustmers:Coustmers
           var mv = ModelAndView("retrive")
           val coustmerlist: List<Coustmers> = repo.findAll() as List<Coustmers>
          // val person:(Coustmers)->Boolean={coustmers:Coustmers->coustmers.cid==(cid)}
           val matchingCoustmer: Optional<Coustmers> =coustmerlist.stream().filter { p->p.cid==(cid) }.findFirst()
           if(matchingCoustmer.isPresent)
           {
               coustmers=matchingCoustmer.get()
               println("successfully found")
               log.info("${coustmers.cusername} successfully  found")
               mv.addObject(coustmers)
           }
           else
           {
               coustmers=matchingCoustmer.orElse(null)
               println("not found")
               log.info("${coustmers.cname} you have entered wrong credentials")

               mv.setViewName("viewcoustmers")
           }
           return mv;
        }


}